cmd/podman/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	bpf "github.com/iovisor/gobpf/bcc"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	seccomp "github.com/seccomp/libseccomp-golang"
+)
+
+type event struct {
+	Pid uint32
+	ID  uint32
+	// Inum    uint
+	Command [16]byte
+}
+
+type calls map[string]int
+
+const source string = `
+#include <linux/bpf.h>
+#include <linux/nsproxy.h>
+#include <linux/pid_namespace.h>
+#include <linux/ns_common.h>
+#include <linux/sched.h>
+#include <linux/tracepoint.h>
+
+BPF_HASH(parent_namespace, u64, unsigned int);
+BPF_PERF_OUTPUT(events);
+
+struct data_t
+{
+    u32 pid;
+    u32 id;
+    char comm[16];
+};
+
+int enter_trace(struct tracepoint__raw_syscalls__sys_enter *args)
+{
+    struct data_t data = {};
+    u64 key = 0;
+    unsigned int zero = 0;
+    struct task_struct *task;
+
+    data.pid = bpf_get_current_pid_tgid();
+    data.id = (int)args->id;
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+
+    task = (struct task_struct *)bpf_get_current_task();
+    struct nsproxy *ns = task->nsproxy;
+    unsigned int inum = ns->pid_ns_for_children->ns.inum;
+
+    if (data.pid == PARENT_PID)
+    {
+        parent_namespace.update(&key, &inum);
+    }
+    unsigned int *parent_inum = parent_namespace.lookup_or_init(&key, &zero);
+
+    if (*parent_inum != inum)
+    {
+        return 0;
+    }
+
+    events.perf_submit(args, &data, sizeof(data));
+    return 0;
+}
+`
+
+func main() {
+	logfile, err := os.OpenFile(os.TempDir()+"/logfile", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		log.Fatalf("error opening file: %v", err)
+	}
+	defer logfile.Close()
+	log.SetOutput(logfile)
+	log.Println(os.Args)
+	terminate := flag.Bool("e", false, "send SIGINT to floating process")
+	runBPF := flag.Int("r", 0, "-r <pid of the container>")
+	flag.Parse()
+
+	if *runBPF > 0 {
```

Comment:
Can we split the major operations here (runBPF, terminate) off into separate functions? Main is very big like this